api/class-wc-rest-dev-mailchimp-settings-controller.php

```diff
@@ -0,0 +1,296 @@
+<?php
+/**
+ * REST API WC MailChimp settings
+ *
+ * Handles requests that interact with MailChimp plugin.
+ *
+ * @author   Automattic
+ * @category API
+ * @package  WooCommerce/API
+ * @since    3.0.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * @package WooCommerce/API
+ */
+class WC_REST_Dev_MailChimp_Settings_Controller extends WC_REST_Dev_Settings_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/v3';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'mailchimp';
+
+	/**
+	 * MailChimp settings.
+	 *
+	 * @param  WP_REST_Request    $request    Request object.
+	 * @return WP_REST_Response   $response   Response data.
+	 */
+
+	/**
+	 * Register MailChimp settings routes
+	 */
+	public function register_routes() {
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_settings' ),
+				'permission_callback' => array( $this, 'update_items_permissions_check' ),
+			)
+		) );
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/api_key', array(
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => array( $this, 'update_api_key' ),
+				'permission_callback' => array( $this, 'update_items_permissions_check' ),
+			),
+		'schema' => array( $this, 'get_api_key_schema' ),
```

Comment:
I don't see a function for this, or `get_store_info_schema` below.

I'm fine leaving the schemas out of here for now since this endpoint is undocumented and for our own purposes at the moment. It would be good to circle back and add them in a separate PR. Could you add an issue for that and then remove these lines since the functions don't exist.

Reply:
I have removed all related schema elements in one of the last commits - missed that one. I was thinking the same that until we have this undocumented there should be nothing visible. Ideally, this should just be part of the plugin. The issue: 
https://github.com/woocommerce/wc-api-dev/issues/62